buildagent-generation-galleryvm.yml

```diff
@@ -11,20 +11,24 @@ parameters:
   displayName: Scripts Path
   type: string
   default: .
+- name: build_stage_job_name
+  displayName: Name of the stage and job that builds the image
+  type: string
+  default: buildagent_template_vm.generate_image
 
 jobs:
 - job: createvmimageversion
   displayName: Create New Gallery VMImage Version
   timeoutInMinutes: 600
   cancelTimeoutInMinutes: 90
-  pool: 
+  pool:
     name: ${{ parameters.agent_pool }}
     
   variables:
   - name: GalleryVmImageDefinition
-    value: $[stageDependencies.buildagent_template_vm.generate_image.outputs['setVars.GalleryVmImageDefinition'] ]
+    value: $[stageDependencies.${{parameters.build_stage_job_name}}.outputs['setVars.GalleryVmImageDefinition'] ]
```

Comment:
Can we add the `.generate_image` back to this line? It's a bit cleaner so we aren't passing that from multiple places.